revapi/src/main/java/org/revapi/DifferenceTransform.java

```diff
@@ -55,6 +58,29 @@ public interface DifferenceTransform<E extends Element<?>> extends AutoCloseable
      */
     Pattern[] getDifferenceCodePatterns();
 
+    /**
+     * Gets the list of {@link Predicate predicates} to match the difference codes this transform can handle.
+     * <p>
+     * This is a replacement for {@link #getDifferenceCodePatterns()} where {@link Predicate} is more generalized and
+     * allows for {@link DifferenceTransform} implementations to use more optimized solutions, such as exact string
+     * matching or custom string matching rather than more performance intensive {@link Pattern}s.
+     *
+     * @return The list of predicates to match the difference codes this transform can handle.
+     */
+    default List<Predicate<String>> getDifferenceCodePredicates() {
+        Pattern[] patterns = getDifferenceCodePatterns();
+        if (patterns == null) {
+            return null;
```

Comment:
Is there a reason for distinguishing between `null` and empty list? If there is no reason for that, I'd rather return an empty list here. If there is a reason, let's annotate the return type as `@Nullable`.

Reply:
I'll change this to be an empty list and marking with `@NonNull`.

I went with this initially as `getDifferenceCodePatterns()` wasn't annotated with anything and made an assumption that nullness was possible. But, given the usage was in a `for (T : T[])` that would've thrown a `NullPointerException`.